notebooks/ssd-object-detection-demo.ipynb

```diff
@@ -0,0 +1,621 @@
+    "## 3. Creating TorchScript modules  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To compile with TRTorch, the model must first be in **TorchScript**. TorchScript is a programming language included in PyTorch which removes the Python dependency normal PyTorch models have. This conversion is done via a JIT compiler which given a PyTorch Module will generate an equivalent TorchScript Module. There are two paths that can be used to generate TorchScript: **Tracing** and **Scripting**. Tracing follows execution of PyTorch generating ops in TorchScript corresponding to what it sees. Scripting does an analysis of the Python code and generates TorchScript, this allows the resulting graph to include control flow which tracing cannot do. Tracing however due to its simplicity is more likely to compile successfully with TRTorch (though both systems are supported)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = ssd300.eval().to(\"cuda\")\n",
+    "traced_model = torch.jit.trace(model, [torch.randn((1,3,300,300)).to(\"cuda\")])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can save this model and use it independently of Python."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# This is just an example, and not required for the purposes of this demo\n",
+    "torch.jit.save(traced_model, \"ssd_300_traced.jit.pt\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Warm up ...\n",
+      "Start timing ...\n",
+      "Iteration 1000/5000, ave batch time 366.47 ms\n",
+      "Iteration 2000/5000, ave batch time 366.43 ms\n",
+      "Iteration 3000/5000, ave batch time 366.42 ms\n",
+      "Iteration 4000/5000, ave batch time 366.43 ms\n",
+      "Iteration 5000/5000, ave batch time 366.42 ms\n",
+      "Input shape: torch.Size([128, 3, 300, 300])\n",
+      "Output location prediction size: torch.Size([128, 4, 8732])\n",
+      "Output label prediction size: torch.Size([128, 81, 8732])\n",
+      "Average batch time: 366.42 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Obtain the average time taken by a batch of input\n",
+    "benchmark(traced_model, input_shape=(128, 3, 300, 300), nruns=5000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "<a id=\"4\"></a>\n",
+    "## 4. Compiling with TRTorch\n",
+    "TorchScript modules behave just like normal PyTorch modules and are intercompatable. From TorchScript we can now compile a TensorRT based module. This module will still be implemented in TorchScript but all the computation will be done in TensorRT."
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
We should clarify that the compiled module will be in FP16 and not FP32. Otherwise the difference in performance does not seem credible and it also it will explain the torch.half

Reply:
Fixed in latest commit.